administrator/components/com_associations/models/associations.php

```diff
@@ -0,0 +1,234 @@
+				'associationcomponent',
+			);
+		}
+
+		parent::__construct($config);
+	}
+
+	/**
+	 * Method to auto-populate the model state.
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @param   string  $ordering   An optional ordering field.
+	 * @param   string  $direction  An optional direction (asc|desc).
+	 *
+	 * @return  void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected function populateState($ordering = 'title', $direction = 'asc')
+	{
+		$this->setState('filter.search', $this->getUserStateFromRequest($this->context . '.filter.search', 'filter_search', '', 'string'));
+		$this->setState(
+			'associationlanguage', $this->getUserStateFromRequest(
+					$this->context . '.associationlanguage', 'associationlanguage', '', 'string'
+				)
+			);
+		$this->setState(
+			'associationcomponent', $this->getUserStateFromRequest(
+					$this->context . '.associationcomponent', 'associationcomponent', '', 'string'
+				)
+			);
+
+		// List state information.
+		parent::populateState($ordering, $direction);
+	}
+
+	/**
+	 * Method to get a store id based on model configuration state.
+	 *
+	 * This is necessary because the model is used by the component and
+	 * different modules that might need different sets of data or different
+	 * ordering requirements.
+	 *
+	 * @param   string  $id  A prefix for the store id.
+	 *
+	 * @return  string  A store id.
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected function getStoreId($id = '')
+	{
+		// Compile the store id.
+		$id .= ':' . $this->getState('filter.search');
+		$id .= ':' . $this->getState('associationlanguage');
+		$id .= ':' . $this->getState('associationcomponent');
+
+		return parent::getStoreId($id);
+	}
+
+	/**
+	 * Build an SQL query to load the list data.
+	 *
+	 * @return  JDatabaseQuery
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected function getListQuery()
+	{
+		// Create a new query object.
+		$db             = $this->getDbo();
+		$query          = $db->getQuery(true);
+		$user           = JFactory::getUser();
+		$app            = JFactory::getApplication();
+		$component      = $this->getState('associationcomponent');
+		$table          = '';
+		$extension      = '';
+
+		// If it's not a category
+		if (!strpos($component, '|'))
+		{
+			$componentSplit = explode('.', $component);
+			$componentModelPath = JPATH_ADMINISTRATOR . '/components/' . $componentSplit[0]	
+				. '/models/' . $componentSplit[1] . '.php';
+			$componentModel     = file_get_contents($componentModelPath);
+
+			if ($position = strpos($componentModel, 'getAssociations'))
+			{
+				// Searching for , '#__table' , after getAssociations(
+				$start = strpos($componentModel, ',', $position) + 2;
+				$end = strpos($componentModel, ',', $start) - 1;
+				
+				if ($componentSplit[0] == 'com_menus')
+				{
+					$table = '#__menu';
+				}
+				else 
+				{
+					$table = str_replace("'", "", substr($componentModel, $start, $end - $start));
+				}
+			}	
```

Comment:
IMO we need a more clean and fail proof method to get the table name.

@infograf768 is there any file where we can get the main component table use din associations? a xml? a class property?